Mathlib/Order/Filter/CardinalInter.lean

```diff
@@ -152,4 +156,188 @@ theorem EventuallyEq.cardinal_bInter {S : Set ι} (hS : #S < c)
   (EventuallyLE.cardinal_bInter hS fun i hi => (h i hi).le).antisymm
     (EventuallyLE.cardinal_bInter hS fun i hi => (h i hi).symm.le)
 
+/-- Construct a filter with cardinal `c` intersection property. This constructor deduces
+`Filter.univ_sets` and `Filter.inter_sets` from the cardinal `c` intersection property. -/
+def ofCardinalInter (l : Set (Set α)) (hc : 2 < c)
+    (hp : ∀ S : Set (Set α), (#S < c) → S ⊆ l → ⋂₀ S ∈ l)
+    (h_mono : ∀ s t, s ∈ l → s ⊆ t → t ∈ l) : Filter α where
+  sets := l
+  univ_sets := by
+    apply @sInter_empty α ▸ hp ∅ (?_) (empty_subset _)
+    have this : 0 < c := lt_trans zero_lt_two hc
+    rwa [mk_eq_zero]
+  sets_of_superset := h_mono _ _
+  inter_sets {s t} hs ht := sInter_pair s t ▸ by
+    apply hp _ (?_) (insert_subset_iff.2 ⟨hs, singleton_subset_iff.2 ht⟩)
+    have : #({s, t} : Set (Set α)) ≤ 2 := by
+      calc
+      _ ≤ #({t} : Set (Set α)) + 1 := Cardinal.mk_insert_le
+      _ = 2 := by norm_num
+    exact lt_of_le_of_lt this hc
+
+instance cardinalInter_ofCardinalInter (l : Set (Set α)) (hc : 2 < c)
```

Comment:
I now realise this can indeed be an instance because the non-typeclass arguments appear in the type of the desired instance. You should change it back: sorry for the bad advice before.